internal/services/eventgrid/eventgrid_partner_namespace_channel_resource.go

```diff
@@ -0,0 +1,455 @@
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.Channels
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var config EventGridPartnerNamespaceChannelResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+			partnerNamespaceId, err := partnernamespaces.ParsePartnerNamespaceID(config.PartnerNamespaceId)
+			if err != nil {
+				return err
+			}
+
+			id := channels.NewChannelID(subscriptionId, partnerNamespaceId.ResourceGroupName, partnerNamespaceId.PartnerNamespaceName, config.ChannelName)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			param := channels.Channel{
+				Name: pointer.To(config.ChannelName),
+				Properties: &channels.ChannelProperties{
+					ChannelType:                     pointer.To(channels.ChannelType(config.ChannelType)),
+					ExpirationTimeIfNotActivatedUtc: pointer.To(config.ExpirationTimeIfNotActivatedInUtc),
+					PartnerTopicInfo:                expandPartnerNamespaceChannelPartnerTopic(config.PartnerTopic),
+				},
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r EventGridPartnerNamespaceChannelResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.Channels
+
+			id, err := channels.ParseChannelID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config EventGridPartnerNamespaceChannelResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `properties` was nil", *id)
+			}
+
+			payload := existing.Model
+
+			// API generated default message can be longer than the allowed length so we have to clear it from update payload
+			payload.Properties.MessageForActivation = nil
+
+			if metadata.ResourceData.HasChange("expiration_time_if_not_activated_in_utc") {
+				if config.ReadinessState == string(channels.ReadinessStateActivated) {
+					return fmt.Errorf("`expiration_time_if_not_activated_in_utc` can not be updated if the channel has been activated")
```

Comment:
could you move this validation into `CustomizeDiff` and use `metadata.ResourceDiff.ForceNew` instead of `fmt.Errorf` ?